libs/methodologies/bold/rule-processors/credits/src/rewards-distribution/rewards-distribution.test-cases.ts

```diff
@@ -0,0 +1,637 @@
+    creditsDocument: documents.multiHauler.creditsDocument,
+    expectedActorsResult: expectedResults.multipleHaulers,
+    expectedCertificateTotalValue: documents.massId1Value,
+    resultStatus: RuleOutputStatus.APPROVED,
+    scenario: 'single certificate with multiple HAULER participants',
+    unitPrice: UNIT_PRICE_VALUE,
+  },
+  {
+    certificateDocuments: documents.multipleCertificates.certificateDocuments,
+    creditsDocument: documents.multipleCertificates.creditsDocument,
+    expectedActorsResult: expectedResults.multipleCertificates,
+    expectedCertificateTotalValue: documents.multipleCertificates.totalValue,
+    resultStatus: RuleOutputStatus.APPROVED,
+    scenario: 'multiple certificates with the same participants',
+    unitPrice: UNIT_PRICE_VALUE,
+  },
+];
+
+const createErrorTestCases = () => {
+  const errorStubs = new BoldStubsBuilder()
+    .createMassIdDocuments()
+    .createMassIdAuditDocuments()
+    .createCertificateDocuments({
+      partialDocument: {
+        type: RECYCLED_ID,
+      },
+    })
+    .createCreditsDocument()
+    .build();
+
+  const creditsDocumentWithoutRulesMetadata = {
+    ...errorStubs.creditsDocument,
+    externalEvents: errorStubs.creditsDocument.externalEvents?.filter(
+      (event) => event.name !== RULES_METADATA.toString(),
+    ),
+  };
+
+  const certificateDocumentWithoutRulesMetadata = {
+    ...errorStubs.certificateDocuments[0]!,
+    externalEvents: errorStubs.certificateDocuments[0]!.externalEvents?.filter(
+      (event) => event.name !== RULES_METADATA.toString(),
+    ),
+  };
+
+  return {
+    certificateDocumentWithoutRulesMetadata,
+    creditsDocumentWithoutRulesMetadata,
+    errorStubs,
+  };
+};
+
+const errorTestData = createErrorTestCases();
+
+export const rewardsDistributionProcessorErrors: ErrorTestCase[] = [
+  {
+    certificateDocuments: [],
+    creditsDocument: errorTestData.errorStubs.creditsDocument,
+    resultComment: ERROR_MESSAGES.CERTIFICATE_DOCUMENT_NOT_FOUND(RECYCLED_ID),
+    resultStatus: RuleOutputStatus.REJECTED,
+    scenario: `the ${RECYCLED_ID} documents is not found`,
+  },
+  {
+    certificateDocuments: [...errorTestData.errorStubs.certificateDocuments],
+    creditsDocument: undefined,
+    resultComment: ERROR_MESSAGES.CREDITS_DOCUMENT_NOT_FOUND,
+    resultStatus: RuleOutputStatus.REJECTED,
+    scenario: `the ${CREDITS} document is not found`,
+  },
+  {
+    certificateDocuments: [...errorTestData.errorStubs.certificateDocuments],
+    creditsDocument: errorTestData.creditsDocumentWithoutRulesMetadata,
+    resultComment: ERROR_MESSAGES.INVALID_UNIT_PRICE,
+    resultStatus: RuleOutputStatus.REJECTED,
+    scenario: `the "${UNIT_PRICE}" attribute in the ${CREDITS} document is invalid`,
+  },
+  {
+    certificateDocuments: [
+      errorTestData.certificateDocumentWithoutRulesMetadata,
+    ],
+    creditsDocument: errorTestData.errorStubs.creditsDocument,
```

Comment:
If we want to be aligned with smaug, we could rename credtis to creditOrder, but feel free to leave it out for now.
```suggestion
    creditOrderDocument: errorTestData.errorStubs.creditOrderDocument,
```

Reply:
https://github.com/carrot-foundation/methodology-rules/pull/222/commits/7a64e08980ee6a4d0c6d63d8b5a9e001e5d99f96